ledger/eval/eval.go

```diff
@@ -1658,78 +1770,26 @@ func (eval *BlockEvaluator) generateKnockOfflineAccountsList() {
 	}
 }
 
-// bitsMatch checks if the first n bits of two byte slices match. Written to
-// work on arbitrary slices, but we expect that n is small. Only user today
-// calls with n=5.
-func bitsMatch(a, b []byte, n int) bool {
-	// Ensure n is a valid number of bits to compare
-	if n < 0 || n > len(a)*8 || n > len(b)*8 {
-		return false
-	}
-
-	// Compare entire bytes when n is bigger than 8
-	for i := 0; i < n/8; i++ {
-		if a[i] != b[i] {
-			return false
-		}
-	}
-	remaining := n % 8
-	if remaining == 0 {
-		return true
-	}
-	return bits.LeadingZeros8(a[n/8]^b[n/8]) >= remaining
-}
+const absentFactor = 20
 
 func isAbsent(totalOnlineStake basics.MicroAlgos, acctStake basics.MicroAlgos, lastSeen basics.Round, current basics.Round) bool {
 	// Don't consider accounts that were online when payouts went into effect as
 	// absent.  They get noticed the next time they propose or keyreg, which
 	// ought to be soon, if they are high stake or want to earn incentives.
-	if lastSeen == 0 {
+	if lastSeen == 0 || acctStake.Raw == 0 {
 		return false
 	}
-	// See if the account has exceeded 10x their expected observation interval.
-	allowableLag, o := basics.Muldiv(10, totalOnlineStake.Raw, acctStake.Raw)
+	// See if the account has exceeded their expected observation interval.
+	allowableLag, o := basics.Muldiv(absentFactor, totalOnlineStake.Raw, acctStake.Raw)
 	if o {
-		// This can't happen with 10B total possible stake, but if we imagine
-		// another algorand network with huge possible stake, this seems reasonable.
+		// This can't happen with 10B total possible stake and a reasonable
+		// absentFactor, but if we imagine another algorand network with huge
+		// possible stake, this seems reasonable.
 		allowableLag = math.MaxInt64 / acctStake.Raw
 	}
 	return lastSeen+basics.Round(allowableLag) < current
```

Comment:
With some degenerate cases (not sure if it's possible with the 10B-total algorand mainnet), if `allowableLag` is very large (close to 2^64), the addition could overflow and lead to a false-positive absent detection.